### PR TITLE
Add script which updates eventuals commit and shallow_since

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
Adds scripts to have the possibility to create auto-PR on every push in main in `3rdparty/eventuals` repo in order to update `eventuals` in the repository needed.
`NOTE:` `make sure` that all branches\refs are tied to main in scripts before merging this PR!!! Also `uncomment all stuff for merge labels`!!!

### CODE REVIEW

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# 'git_repository' rule for eventuals in current $file.
+git_repository_needed=False
+
+# Flags to help understand whether or not all updates
+# (commit and shallow_since) are done.
+is_commit_replaced=False
+is_shallow_replaced=False
+
+while IFS= read -r line
+do
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:39]: Let's keep the formatting consistent and we put things like `then` on the same line as `if`:

```suggestion
while IFS= read -r line; do
```

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  # Check if current line of the file contains necessary name
+  # of the git_repository which we gonna update with commit
+  # and shallow_since.
+  if [[ "${line}" =~ .*"name = \"com_github_3rdparty_eventuals\"".* ]];
+  then
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:44]:
```suggestion
  if [[ "${line}" =~ .*"name = \"com_github_3rdparty_eventuals\"".* ]]; then
```

Here and everywhere else!

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+fi
+
+if [[ ${#shallow_since} == 0 ]]; then
+    echo "Incorrect shallow_since value."
+    exit 1
+fi
+
+# This flag indicates whether or not we process correct
+# 'git_repository' rule for eventuals in current $file.
+git_repository_needed=False
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:31]:
```suggestion
# This variable indicates whether or not we have found the correct
# 'git_repository' rule for eventuals in current $file.
git_repository_found=False
```

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+# The latest shallow_since for eventuals.
+shallow_since=$3
+
+if [[ ! -f "$file" ]]; then
+    echo "$file doesn't exist"
+    exit 1
+fi
+
+if [[ ${#commit} == 0 ]]; then
+    echo "Incorrect commit value."
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:20]: Let's be consistent no periods!
```suggestion
    echo "Missing or invalid commit value"
```

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    exit 1
+fi
+
+if [[ ${#shallow_since} == 0 ]]; then
+    echo "Incorrect shallow_since value."
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:25]:
```suggestion
    echo "Missing or invalid shallow_since value"
```

File: .github/workflows/scripts/update-eventuals-git-repository.sh [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    exit 1
+fi
+
+# This flag indicates whether or not we process correct
+# 'git_repository' rule for eventuals in current $file.
+git_repository_needed=False
+
+# Flags to help understand whether or not all updates
```
Comment (benh) [inline .github/workflows/scripts/update-eventuals-git-repository.sh:33]:
```suggestion
# Variables to help understand whether or not all updates
```

File: .github/workflows/update-eventuals-bot.yml [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+      # "Create Pull Request or update existing" in order to merge the changes in automatic mode.
+      # See docs of the action: https://github.com/actions-ecosystem/action-add-labels
+      # - name: Add label "mergequeue-ready"
+      #   if: steps.pr.outputs.pull-request-operation == 'created' || steps.pr.outputs.pull-request-operation == 'updated'
+      #   uses: actions-ecosystem/action-add-labels@v1
+      #   with:
+      #     number: ${{ steps.pr.outputs.pull-request-number }}
+      #     labels: mergequeue-ready
+      #     repo: ${{ inputs.owner }}/${{ inputs.repo_name }}
+      #     github_token: ${{ secrets.private_repo_access_as_rebot_token }}
```
Comment (benh) [inline .github/workflows/update-eventuals-bot.yml:157]: If you don't need this we should remove it!

**Review by benh — CHANGES_REQUESTED** [review]
Since this is eventuals specific lets move this all into the eventuals repo please!

File: .github/workflows/update-eventuals-bot.yml (hunk 7 above)
Comment (ArthurBandaryk) [reply .github/workflows/update-eventuals-bot.yml:157]: We need this for auto-merging. I commented this stuff in order to not make this PR do any auto-merge in eventuals-tutorial before you review it. So, I will uncomment this before I merge it.

**Comment by ArthurBandaryk** [comment]
Since all this stuff was moved to `eventuals` repo (check [this](https://github.com/3rdparty/eventuals/pull/402) PR), we close this!

**Pull request closed without merging**